packages/components/src/components/hds/text/body.ts

```diff
@@ -0,0 +1,99 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { assert } from '@ember/debug';
+import { HdsTextSizeValues, HdsTextWeightValues } from './types.ts';
+import type {
+  HdsTextAligns,
+  HdsTextColors,
+  HdsTextTags,
+  HdsTextWeights,
+} from './types.ts';
+
+// notice: only some combinations of size + font-weight are allowed (per design specs)
+// see: https://www.figma.com/file/oQsMzMMnynfPWpMEt91OpH/HDS-Product---Foundations?node-id=1262%3A9192
+
+// Allow consumers to provide either string or number representation of size
+type HdsTextBodySizeNumber = Extract<HdsTextSizeValues, HdsTextSizeValues.OneHundred | HdsTextSizeValues.TwoHundred | HdsTextSizeValues.ThreeHundred>;
+type HdsTextBodySizeString = `${HdsTextBodySizeNumber}`;
+export type HdsTextBodySizes = HdsTextBodySizeNumber | HdsTextBodySizeString;
+export const AVAILABLE_SIZES = [HdsTextSizeValues.ThreeHundred, HdsTextSizeValues.TwoHundred, HdsTextSizeValues.OneHundred];
+export const DEFAULT_SIZE = HdsTextSizeValues.TwoHundred;
+
+export const DEFAULT_WEIGHT = HdsTextWeightValues.Regular;
+export type HdsTextBodyWeight = Extract<HdsTextWeights, "regular" | "medium" | "semibold">;
+export const AVAILABLE_WEIGHTS_PER_SIZE: Record<HdsTextBodySizes, HdsTextBodyWeight[]> = {
+  300: [HdsTextWeightValues.Regular, HdsTextWeightValues.Medium, HdsTextWeightValues.Semibold],
+  200: [HdsTextWeightValues.Regular, HdsTextWeightValues.Medium, HdsTextWeightValues.Semibold],
+  100: [HdsTextWeightValues.Regular, HdsTextWeightValues.Medium, HdsTextWeightValues.Semibold],
+};
+
+export interface HdsTextBodySignature {
+  Args: {
+    size: HdsTextBodySizes;
```

Comment:
Since we have a predefined value, this can be optional as well, right?
```suggestion
    size?: HdsTextBodySizes;
```

Reply:
💯 good catch, updated in the three subcomponents!